src/eve/concepts.py

```diff
@@ -170,7 +177,13 @@ def __new__(mcs, name, bases, namespace, **kwargs):
         return cls
 
 
-class BaseNode(BaseModel, metaclass=NodeMetaclass):
+AnyNode = TypeVar("AnyNode", bound="BaseNode")
+AnyNodeLeaf = Union[bool, bytes, int, float, str, IntEnum, StrEnum, AnyNode]
```

Comment:
Isn't the `Any` prefix superfluous? I like `LeafNode`.

Reply:
`Any` is superfluous, I agree, and it could be removed. It is only used as a hint that this is a typevar. Would it be better to use `AnyLeafNode` or should we just use `LeafNode` as you propose?

Reply:
Renamed to LeafNode